src/main/java/com/example/demo/webclient/OpenWeatherServiceRestImpl.java

```diff
@@ -7,10 +7,16 @@
 import org.springframework.web.client.RestTemplate;
 import org.springframework.web.util.UriComponentsBuilder;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
 @Service
 @Slf4j
 public class OpenWeatherServiceRestImpl {
     private static final String BASE_URL = "https://api.openweathermap.org/data/2.5/forecast";
+    private static final String GEO_URL = "https://api.openweathermap.org/geo/1.0/direct";
+
     private final RestTemplate restTemplate = new RestTemplate();
```

Comment:
```suggestion
public class OpenWeatherServiceRestImpl {
    private static final String BASE_URL = "/data/2.5/forecast";
    private static final String GEO_URL = "/geo/1.0/direct";
    private final RestTemplate restTemplate = new RestTemplateBuilder()
    .rootUri("https://api.openweathermap.org")
    .build();
```
можна зробити якось так, щоб не повторювати всюду сталу частину URL